### PR TITLE
Add drawer for node edition

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

File: src/components/graph/menus/editable-title.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { darken, makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+    header: {
+        backgroundColor: darken(theme.palette.background.default, 0.2),
+    },
+}));
+
+export const EditableTitle = ({ name, onClose, onChange }) => {
+    const [editTitle, setEditTitle] = useState(false);
```
Comment (sBouzols) [inline src/components/graph/menus/editable-title.js:22]: `openEditTitle` ? to be homogenous with others states which manage open/close dialogs

File: src/components/graph/menus/editable-title.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                    <EditIcon />
+                </IconButton>
+                <span style={{ flexGrow: '1' }}>{name}</span>
+                <IconButton size={'small'} onClick={onClose}>
+                    <CloseIcon />
+                </IconButton>
+            </Typography>
+            <AskTextDialog
+                show={editTitle}
+                title={'changeName'}
```
Comment (sBouzols) [inline src/components/graph/menus/editable-title.js:46]: Title should be translated

File: src/components/util/ask-text-dialog.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import {
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    TextField,
+} from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+
+export const AskTextDialog = ({ title, value, show, onValidate, onClose }) => {
```
Comment (sBouzols) [inline src/components/util/ask-text-dialog.js:18]: Add JSDoc

File: src/components/util/ask-text-dialog.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                    {intl.formatMessage({ id: 'validate' })}
+                </Button>
+                <Button color={'primary'} onClick={onClose}>
+                    {intl.formatMessage({ id: 'cancel' })}
+                </Button>
+            </DialogActions>
+        </Dialog>
+    );
+};
+
```
Comment (sBouzols) [inline src/components/util/ask-text-dialog.js:51]: Add propsType

File: src/components/graph/menus/node-editor.js [hunk 5]
```diff
@@ -5,41 +5,101 @@
...
+                displayErrorMessageWithSnackbar({
+                    errorMessage: err.message,
+                    enqueueSnackbar,
+                })
+            );
+    }, [setSelectedNode, enqueueSnackbar, selectedNodeUuid]);
+
+    const changeNodeName = (newName) => {
+        updateTreeNode({
+            id: selectedNode?.id,
```
Comment (sBouzols) [inline src/components/graph/menus/node-editor.js:61]: Why do you check selectedNode definition here ?
If selectedNode is undefined here, maybe you should completly avoid the updateTreeNode ?

File: src/components/graph/menus/node-editor.js [hunk 6]
```diff
@@ -5,41 +5,101 @@
...
+import {
+    displayErrorMessageWithSnackbar,
+    useIntlRef,
+} from '../../../utils/messages';
+import { EditableTitle } from './editable-title';
+import { useSnackbar } from 'notistack';
+import { useSelector } from 'react-redux';
 
 const useStyles = makeStyles((theme) => ({
     root: {
```
Comment (sBouzols) [inline src/components/graph/menus/node-editor.js:27]: seems not used ?

File: src/components/graph/menus/node-editor.js [hunk 7]
```diff
@@ -5,41 +5,101 @@
...
+    };
 
     return (
-        <Paper className={classes.paper}>
-            {selectedNode.type === 'NETWORK_MODIFICATION' && (
-                <NetworkModificationNodeEditor selectedNode={selectedNode} />
-            )}
-            {selectedNode.type === 'MODEL' && (
-                <ModelNodeEditor selectedNode={selectedNode} />
+        <div {...props}>
```
Comment (sBouzols) [inline src/components/graph/menus/node-editor.js:77]: Why do you pass props to div here ? is it not a bit dangerous or too flexible ?

File: src/components/graph/menus/network-modification-node-editor.js [hunk 8]
```diff
@@ -5,66 +5,107 @@
...
+import { ListItem } from '@material-ui/core';
+import { FormattedMessage } from 'react-intl';
+import Divider from '@material-ui/core/Divider';
 
-const NetworkModificationNodeEditor = ({ selectedNode }) => {
-    const intl = useIntl();
-    const intlRef = useIntlRef();
+const useStyles = makeStyles((theme) => ({
+    list: {
+        flexGrow: '1',
```
Comment (sBouzols) [inline src/components/graph/menus/network-modification-node-editor.js:26]: You can't be sure that your parent is a flex container then this css could be useless.
Maybe we should have a parent flex div in this component which makes this consistent.

**Review by sBouzols — CHANGES_REQUESTED** [review]
Some remarks
I must test tomorrow

File: src/components/graph/menus/modification-list-item.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    );
+    return (
+        <>
+            <ListItem {...props}>
+                <OverflowableText text={getLabel()} />
+            </ListItem>
+            <Divider />
+        </>
+    );
+};
```
Comment (FranckLecuyer) [inline src/components/graph/menus/modification-list-item.js:41]: Missing PropTypes

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/translations/network-modifications-fr.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
+{
+    "network_modification/modificationsCount": "{count, plural, =0 {aucune modification} =1 {# modification} other {# modifications}}",
+    "network_modifications/EQUIPMENT_DELETION": "Suppression de\n {computedLabel}",
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/translations/network-modifications-fr.json:3]: removed

File: src/components/graph/menus/editable-title.js (hunk 1 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/editable-title.js:22]: done

File: src/components/graph/menus/editable-title.js [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+    },
+}));
+
+export const EditableTitle = ({ name, onClose, onChange }) => {
+    const [editTitle, setEditTitle] = useState(false);
+    const classes = useStyles();
+
+    return (
+        <div>
+            <Typography
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/graph/menus/editable-title.js:27]: done

File: src/components/graph/menus/editable-title.js (hunk 2 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/editable-title.js:46]: done and changed to NewName

File: src/components/util/ask-text-dialog.js (hunk 3 above)
Comment (JacquesBorsenberger) [reply src/components/util/ask-text-dialog.js:18]: done

File: src/components/util/ask-text-dialog.js (hunk 4 above)
Comment (JacquesBorsenberger) [reply src/components/util/ask-text-dialog.js:51]: done

File: src/components/graph/menus/node-editor.js (hunk 5 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/node-editor.js:61]: done

File: src/components/graph/menus/node-editor.js (hunk 6 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/node-editor.js:27]: removed

File: src/translations/network-modifications-fr.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+    "network_modifications/LOAD_CREATION": "Création de la charge\n {computedLabel}",
+    "network_modifications/GENERATOR_CREATION": "Création du generateur\n {computedLabel}",
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/translations/network-modifications-fr.json:5]: done

File: src/components/graph/menus/editable-title.js [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+                title={'changeName'}
+                value={name}
+                onValidate={(e) => {
+                    onChange(e);
+                }}
+                onClose={() => setEditTitle(false)}
+            />
+        </div>
+    );
+};
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/graph/menus/editable-title.js:55]: done

File: src/components/graph/menus/modification-list-item.js (hunk 9 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/modification-list-item.js:41]: done

File: src/components/graph/menus/modification-list-item.js [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+import { useSelector } from 'react-redux';
+import { PARAM_USE_NAME } from '../../../utils/config-params';
+import Divider from '@material-ui/core/Divider';
+
+export const ModificationListItem = ({ modification, ...props }) => {
+    const intl = useIntl();
+    const useName = useSelector((state) => state[PARAM_USE_NAME]);
+
+    const getComputedLabel = useCallback(() => {
+        if (useName && modification.equipmentName)
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/components/graph/menus/modification-list-item.js:20]: done

File: src/components/graph/menus/network-modification-node-editor.js (hunk 8 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/network-modification-node-editor.js:26]: the parent still need to grow, if the parent is not flex, the added div will not grow

File: src/components/graph/menus/node-editor.js (hunk 7 above)
Comment (JacquesBorsenberger) [reply src/components/graph/menus/node-editor.js]: changed, but I don't think flexibility is not a bad thing.

File: src/components/graph/menus/network-modification-node-editor.js [hunk 15]
```diff
@@ -5,66 +5,103 @@
...
     return (
         <>
-            <Grid container spacing={3} alignItems="flex-end">
-                <Grid item xs={12}>
-                    <TextField
-                        label={intl.formatMessage({ id: 'Name' })}
-                        value={nameValue}
-                        onChange={updateNameValue}
+            <List className={classes.list}>
+                <ListItem>
```
Comment (sBouzols) [inline src/components/graph/menus/network-modification-node-editor.js:80]: This ListItem should have a key too .

**Review by sBouzols — APPROVED** [review]
Tests ok
some functionalities unavailable but it needs other devs.

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]
Label not changed in network-modifications-fr.json
for two windings transformer creation

File: src/translations/network-modifications-fr.json [hunk 16]
```diff
@@ -0,0 +1,8 @@
...
+    "network_modifications/LINE_CREATION": "Création de la ligne\n {computedLabel}",
+    "network_modifications/TWO_WINDINGS_TRANSFORMER_CREATION": "Création du transformateur 2 enroulements\n {computedLabel}"
```
[the remark this answers is not in the document]
Comment (JacquesBorsenberger) [reply src/translations/network-modifications-fr.json]: you're looking on the old changeset, the change is there

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**